### ISSUE TITLE
Apparent regex failure for three commas

### ISSUE THREAD

**Issue by michaeltlombardi** [jdkato/txtlint#1038]
### Check for existing issues

- [x] Completed

### Environment

- OS: Windows 11
- Install: Chocolatey
- Vale version: `3.12.0`

### Describe the bug / provide steps to reproduce it

I've been trying to write a rule to catch repeated consecutive commas, but the rule seems to catch 2 commas and 4 or more commas, but not three commas. I can't really think of a reason _why_ three commas isn't triggering the rule but other amounts are.

<details open><summary>Rule definition</summary>

```yaml
# en_US_punctation.avoid_consecutive_commas.yml
extends: substitution
level: error
nonword: true
description: >-
  Multiple consecutive commas have no meaningful semantics in US English. If
  the comma was accidentally repeated, it should be replaced with a single
  comma. If the commas were accidentally used instead of periods for an
  ellipsis (`...`), it should be replaced with the ellipsis.
message: >-
  Use a single comma (`,`) or an ellipsis (...), not multiple consecutive
  commas.
action:
  name: replace
swap:
  - ',{2,}': ',|...'
```

</details>

<details open><summary>Test file</summary>

```markdown
<!-- test.md --> <!-- markdownlint-disable -->
One, comma
Two,, commas
Three,,, commas
Four,,,, commas
Five,,,,, commas
Six,,,,,, commas
```

</details>

<details><summary>Repro test results</summary>

```sh
vale ./test.md --output JSON
```

```json
{
  "test.md": [
    {
      "Action": {
        "Name": "replace",
        "Params": [
          ",",
          "..."
        ]
      },
      "Span": [
        4,
        5
      ],
      "Check": "en_US_punctuation.avoid_consecuritve_commas",
      "Description": "Multiple consecutive commas have no meaningful semantics in US English. If the comma was accidentally repeated, it should be replaced with a single comma. If the commas were accidentally used instead of periods for an ellipsis (`...`), it should be replaced with the ellipsis.",
      "Link": "",
      "Message": "Use a single comma (`,`) or an ellipsis (...), not multiple consecutive commas.",
      "Severity": "error",
      "Match": ",,",
      "Line": 3
    },
    {
      "Action": {
        "Name": "replace",
        "Params": [
          ",",
          "..."
        ]
      },
      "Span": [
        5,
        8
      ],
      "Check": "en_US_punctuation.avoid_consecuritve_commas",
      "Description": "Multiple consecutive commas have no meaningful semantics in US English. If the comma was accidentally repeated, it should be replaced with a single comma. If the commas were accidentally used instead of periods for an ellipsis (`...`), it should be replaced with the ellipsis.",
      "Link": "",
      "Message": "Use a single comma (`,`) or an ellipsis (...), not multiple consecutive commas.",
      "Severity": "error",
      "Match": ",,,,",
      "Line": 5
    },
    {
      "Action": {
        "Name": "replace",
        "Params": [
          ",",
          "..."
        ]
      },
      "Span": [
        5,
        9
      ],
      "Check": "en_US_punctuation.avoid_consecuritve_commas",
      "Description": "Multiple consecutive commas have no meaningful semantics in US English. If the comma was accidentally repeated, it should be replaced with a single comma. If the commas were accidentally used instead of periods for an ellipsis (`...`), it should be replaced with the ellipsis.",
      "Link": "",
      "Message": "Use a single comma (`,`) or an ellipsis (...), not multiple consecutive commas.",
      "Severity": "error",
      "Match": ",,,,,",
      "Line": 6
    },
    {
      "Action": {
        "Name": "replace",
        "Params": [
          ",",
          "..."
        ]
      },
      "Span": [
        4,
        9
      ],
      "Check": "en_US_punctuation.avoid_consecuritve_commas",
      "Description": "Multiple consecutive commas have no meaningful semantics in US English. If the comma was accidentally repeated, it should be replaced with a single comma. If the commas were accidentally used instead of periods for an ellipsis (`...`), it should be replaced with the ellipsis.",
      "Link": "",
      "Message": "Use a single comma (`,`) or an ellipsis (...), not multiple consecutive commas.",
      "Severity": "error",
      "Match": ",,,,,,",
      "Line": 7
    }
  ]
}
```

</details>

**Comment by ccoVeille** [comment]
Hi @michaeltlombardi 

Did you try playing with nonwords flag ?

https://vale.sh/docs/checks/substitution

**Comment by michaeltlombardi** [comment]
> Hi [@michaeltlombardi](https://github.com/michaeltlombardi)
> 
> Did you try playing with nonwords flag ?
> 
> https://vale.sh/docs/checks/substitution

Yup! I defined the rule with `nonword: true` and that's the definition in the original post - that allowed me to swap two commas and 4+ commas, but not 3 commas.
